tests/backup/backup_restore_basic_test.go

```diff
@@ -1068,3 +1069,209 @@ var _ = Describe("{AllNSBackupWithIncludeNewNSOption}", func() {
+			for _, provider := range providers {
+				customBackupLocationName = fmt.Sprintf("autogenerated-backup-location-%v", time.Now().Unix())
+				err := CreateBackupLocation(provider, customBackupLocationName, backupLocationUID, credName, cloudCredUID, getGlobalBucketName(provider), orgID, "")
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Creating backup location %s", customBackupLocationName))
+			}
+		})
+		Step("Taking backup of applications to trigger BackupSync goroutine", func() {
+			log.InfoD("Taking backup of applications to trigger BackupSync goroutine")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx failed")
+			for _, namespace := range bkpNamespaces {
+				backupName = fmt.Sprintf("%s-%s-%v", BackupNamePrefix, namespace, time.Now().Unix())
+				backupNamespaceMap[namespace] = backupName
+				err = CreateBackup(backupName, SourceClusterName, customBackupLocationName, backupLocationUID, []string{namespace}, labelSelectors, orgID, clusterUid, "", "", "", "", ctx)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying backup creation %s", backupName))
+				backupNames = append(backupNames, backupName)
+			}
+		})
+
+		Step("Check if all backups are synced or not", func() {
+			log.InfoD("Check if backups created before are synced or not")
+
+			// Wait for backups to get synced
+			checkBackupSync := func() (interface{}, bool, error) {
+				fetchedBackupNames, err := GetAllBackupsAdmin()
+				if err != nil {
+					return "", true, fmt.Errorf("unable to fetch backups")
+				}
+				if len(fetchedBackupNames) == len(backupNames) {
+					return "", false, nil
+				}
+				return "", true, fmt.Errorf("expected: %d and actual: %d", len(backupNames), len(fetchedBackupNames))
+			}
+			_, err := task.DoRetryWithTimeout(checkBackupSync, 100*time.Minute, 30*time.Second)
+			log.FailOnError(err, "Wait for BackupSync to complete")
+			fetchedBackupNames, err := GetAllBackupsAdmin()
+			log.FailOnError(err, "Getting a list of all backups")
+			dash.VerifyFatal(len(fetchedBackupNames), len(backupNames), "Comparing the expected and actual number of backups")
+			var bkp *api.BackupObject
+			backupDriver := Inst().Backup
+			bkpEnumerateReq := &api.BackupEnumerateRequest{
+				OrgId: orgID}
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx failed")
+			curBackups, err := backupDriver.EnumerateBackup(ctx, bkpEnumerateReq)
+			for _, bkp = range curBackups.GetBackups() {
+				backupInspectRequest := &api.BackupInspectRequest{
+					Name:  bkp.Name,
+					Uid:   bkp.Uid,
+					OrgId: orgID,
+				}
+				resp, err := backupDriver.InspectBackup(ctx, backupInspectRequest)
+				log.FailOnError(err, "Inspect each backup from list")
+				actual := resp.GetBackup().GetStatus().Status
+				expected := api.BackupInfo_StatusInfo_Success
+				dash.VerifyFatal(actual, expected, "Check each backup for success status")
+			}
+		})
+	})
+	JustAfterEach(func() {
```

Comment:
Are the backups we created being deleted?

Reply:
Yes. If you give `DeleteBackupLocation()` with true it will remove all the backups.